### PR TITLE
fix usage of project search in find_collection function

### PR DESCRIPTION
**Pull request by zoesyang**
# Citrine Python PR

## Description 
Updates the find_collection function to correctly use search parameters. This should help speed up tests.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/seeding/find_or_create.py [hunk 1]
```diff
@@ -15,9 +15,11 @@ def find_collection(collection, name):
             # call list() to collapse the iterator, otherwise the NotFound
             # won't show up until collection_list is used
             collection_list = list(collection.search(search_params={
-                "name": {
-                    "value": name,
-                    "search_method": "EXACT"
+                "search_params": {
```
Comment (maxhutch) [inline src/citrine/seeding/find_or_create.py:18]: I think nesting a "search_params" key inside an argument called "search_params" is very confusing UX.  Could this be fixed by adjusting the logic in the search method [here](https://github.com/CitrineInformatics/citrine-python/blob/master/src/citrine/resources/project.py#L469) ?

Comment (zoesyang) [reply src/citrine/seeding/find_or_create.py:18]: In retrospect, I probably should have changed the ux for that - i was keeping it the same as what the product api request expects. I'll change it

**Comment by zoesyang** [comment]
@maxhutch I changed the search_params to not include 'search_params' as a key. test_experimental.py::test_not_experimental e2e for orion should trigger the search

File: src/citrine/resources/project.py [hunk 2]
```diff
@@ -392,19 +392,17 @@ def search(self, search_params: Optional[dict] = None,
...
+                    "search_method": "EXACT"
+                },
+                "description": {
+                    "value": "polymer chain length",
+                    "search_method": "SUBSTRING"
+                },
             }
             The dict can constain any combination of (one or all) search specifications for the
-            name, description, and status fields of a project. For each parameter specified, the
+        name, description, and status fields of a project. For each parameter specified, the
```
Comment (maxhutch) [inline src/citrine/resources/project.py:405]:
```suggestion
            name, description, and status fields of a project. For each parameter specified, the
```

Comment (maxhutch) [reply src/citrine/resources/project.py:405]: This change looks accidental.

Comment (zoesyang) [reply src/citrine/resources/project.py]: It is! just pushed change

**Comment by zoesyang** [comment]
> The client behavior looks great. Thanks!
> 
> > test_experimental.py::test_not_experimental e2e for orion should trigger the search
> 
> It does hit this route, but it doesn't ever test that the search only returns matching records. I'd guess that's how this bug was caught: there were calls that we thought would be lightweight (because they only returned a few matching records) that were actually returning everything that list would.
> 
> Could we add an e2e that asserts that only matching results are returned?

No problem! I can add an e2e for both the search functionality and the find_collection function in devkit once this is merged :)

**Comment by zoesyang** [comment]
> I verified the basic change locally in lieu of an existing e2e:
> 
> ```
> In [3]: res = list(client.projects.search({"name": {"value": "[PRELOAD", "search_method": "SUBSTRING"}})
>    ...: )                                                                                               
> 
> In [4]: len(res)                                                                                        
> Out[4]: 2
> 
> In [5]: all = list(client.projects.search())                                                            
> 
> In [6]: len(all)                                                                                        
> Out[6]: 289
> ```

Do you still think it would be a good idea to have an e2e?

**Comment by maxhutch** [comment]
> Do you still think it would be a good idea to have an e2e? Also thank you for verifying!

Yeah, especially because it should be fast.  I'm not sure what the backend test coverage of the different search options is, but the different fields and search methods should be checked _somewhere_.

**Pull request merged**